src/main/java/org/gridsuite/study/server/NetworkModificationTreeService.java

```diff
@@ -252,4 +253,80 @@ public void doUpdateNode(AbstractNode node) {
             return node;
         });
     }
+
+    public UUID getStudyRootNodeUuid(UUID studyId) {
+        List<NodeEntity> nodes = nodesRepository.findAllByStudyId(studyId);
+        if (nodes.isEmpty()) {
+            throw new StudyException(ELEMENT_NOT_FOUND);
+        }
+        return nodes.stream().filter(n -> n.getType().equals(NodeType.ROOT)).findFirst().orElseThrow().getIdNode();
+    }
+
+    @Transactional
+    public Optional<String> doGetVariantId(UUID id, boolean generateId) {
```

Comment:
Is the boolean parameter `generateId` necessary since it is set to `true` in all calls to `doGetVariantId` ?

Reply:
doGetVariantId can be further called with false as second parameter, when we will only
want to get the existing variant id associated to a node (when deleting nodes for example, to remove the network variant)

Reply:
Ok